bindings/java/src/org/sleuthkit/datamodel/HashUtility.java

```diff
@@ -42,6 +42,19 @@
 	 * @return md5 of the given FsContent object
 	 */
 	static public String calculateMd5(AbstractFile file) throws IOException {
+		return calculateMd5(file, true);
+	}
+	
+	/**
+	 * Calculate the MD5 hash for the given FsContent and optionally store it in the
+	 * database. The hash will always be written to the AbstractFile object.
+	 *
+	 * @param file file object whose md5 hash we want to calculate
+	 * @param saveToDatabase true to save the hash to the database immediately
+	 *
+	 * @return md5 of the given FsContent object
+	 */
+	static public String calculateMd5(AbstractFile file, boolean saveToDatabase) throws IOException {
```

Comment:
I would like to handle this API change a little bit differently. Let's create a new calculateMD5Hash method that takes a Content instead of an AbstractFile, calculates the hash, and returns it. Then let's deprecate the existing calculateMd5 method, which should call the new method and then save the result. Thanks!

Reply:
Content doesn't have a getSleuthkitCase() method

Reply:
Hmmm, of course not. Let's cast to AbstractContent and throw an exception if instanceof fails.

Reply:
Okay, after in-person discussion, going with Content is more work than I expected. So we are going to stick with AbstractFile.